corehq/apps/domain/tests/test_views.py

```diff
@@ -540,6 +543,100 @@ def test_confirm_renewal_page_context(self):
         self.assertEqual(response.context['next_plan'], expected_next_plan.user_facing_description)
 
 
+class TestCredentialsApplicationSettingsView(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.domain = generator.arbitrary_domain()
+        self.user = generator.arbitrary_user(self.domain.name, is_webuser=True, is_admin=True)
+        self.account = generator.billing_account(self.user, self.user.name)
+        self.client = Client()
+        self.client.force_login(user=self.user.get_django_user())
+
+        self.url = reverse(CredentialsApplicationSettingsView.urlname, args=[self.domain.name])
+
+    def tearDown(self):
+        self.user.delete(self.domain.name, deleted_by=None)
+        self.domain.delete()
+        super().tearDown()
+
+    def test_create_new_credential_app(self):
+        app_id = self._get_app().id
+
+        self.client.post(
+            self.url,
+            data={
+                'app_id': app_id,
+            }
+        )
+        credential_app = CredentialApplication.objects.get(app_id=app_id)
+        assert credential_app.activity_level == CredentialApplication.ActivityLevelChoices.THREE_MONTHS
+
+        app = get_app(self.domain.name, app_id)
+        assert app.profile['features']['credentials'] == CredentialApplication.ActivityLevelChoices.THREE_MONTHS
+
+    def test_specify_new_credential_app_with_existing_app(self):
+        app_id = self._get_app().id
+        self.client.post(
+            self.url,
+            data={
+                'app_id': app_id,
```

Comment:
nit: It would be nice to split the view into a unit that's responsible for 'view' and another method that handles app-setup. With that we wouldn't need to have all the verbosity of client/post/data. And, you could DRY this up in the tests here itself as well.

Not blocking at all, though